src/ReactionHandler.ts

```diff
@@ -84,16 +88,22 @@ export default class ReactionHandler {
    * @param member The guild member the selector is created for
    * @returns A message action row array containing all selectable roles
    */
-  private generateRoleSelectorComponent(member: GuildMember | null): MessageActionRow[] {
+  private async generateRoleSelectorComponent(
+    guild: string | null,
+    member: GuildMember | null
+  ): Promise<MessageActionRow[]> {
     const messageActionRows: MessageActionRow[] = [];
     let currentMessageActionRow: MessageActionRow;
+    if (guild === null) return [];
+
+    const roles = (await GuildSettings.settings(guild)).roles;
```

Comment:
```suggestion
    const { roles } = (await GuildSettings.settings(guild));
```

Reply:
Thinking about it, I would rather prefer 
```suggestion
    const roles: RoleField[] = (await GuildSettings.settings(guild)).roles;
```
since I want to strongly type everything and when I only want to receive one variable, the ` { … } ` syntax doesn't look that much better